### PR TITLE
Add some function or method docstring

### PR DESCRIPTION
**Pull request by dota17**
No big changes, just add some missing function/method docstring.
Cc: @takluyver

### CODE REVIEW

File: h5py/_hl/dataset.py [hunk 1]
```diff
@@ -989,10 +991,12 @@ def flush(self):
         @property
         @with_phil
         def is_virtual(self):
+            """Check if a dataset creation property list layout is virtual"""
```
Comment (takluyver) [inline h5py/_hl/dataset.py:994]:
```suggestion
            """Check if this is a virtual dataset"""
```

File: h5py/_hl/dataset.py [hunk 2]
```diff
@@ -989,10 +991,12 @@ def flush(self):
...
             return self._dcpl.get_layout() == h5d.VIRTUAL
 
         @with_phil
         def virtual_sources(self):
+            """Get all dataset creation property list virtual sources if they exist"""
```
Comment (takluyver) [inline h5py/_hl/dataset.py:999]:
```suggestion
            """Get a list of the data mappings for a virtual dataset"""
```

File: h5py/_hl/files.py [hunk 3]
```diff
@@ -50,6 +51,7 @@ def _set_fapl_mpio(plist, **kwargs):
 
 
 def _set_fapl_fileobj(plist, **kwargs):
+    """Set file access property list for file objects"""
```
Comment (takluyver) [inline h5py/_hl/files.py:54]:
```suggestion
    """Set the Python file object driver in a file access property list"""
```

File: h5py/_hl/dataset.py [hunk 4]
```diff
@@ -524,10 +524,12 @@ def fillvalue(self):
     @cached_property
     @with_phil
     def _extent_type(self):
+        """Get simple extent type for this dataset"""
```
Comment (takluyver) [inline h5py/_hl/dataset.py:527]:
```suggestion
        """Get extent type for this dataset - SIMPLE, SCALAR or NULL"""
```

**Comment by dota17** [comment]
Thanks for your patience!

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**